// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
//import "google/protobuf/empty.proto";
// import "google/protobuf/struct.proto";

package lidar.collector;

// Interface exported by the server.
service DataService {
  // 查询 模板
  rpc GetLidarData (DataRequest_) returns (DataResult_) {}
  
}


////////////////////////////////////////////////////////////////////////////////

message DataRequest_ {
  int64 timestamp = 1;
  int32 step = 2;    // camera alias
  string filters = 3; // 
}

message DataResult_ {

  // 0 表示成功，
  // ！0 表示失败， 失败代码在定义
  // 非 0 时， missions 为空，无效
  int32 status = 1; // 0, 1 ,

  // 如果错误， message 额外的描述信息
  string message = 2;

  string details = 3;

  bytes  data = 4; // camera alias

}


 